File: TextAnalyser/ClientApp/src/app/Models/SemanticModel.ts
import { SemanticCore } from "./SemanticCore";
import { Readability } from "./Readability";

export interface SemanticModel {
    charCount: number,
    charCountWithoutSpaces: number,
    wordCount: number,
    unicWordCount: number,
    significantWordCount: number,
    stopWordCount: number,
    senctenceCount: number,
    waterPercentage: number,
    grammaticalErrorsCount: number,
    clasicNauseaPercentage: number,
    academicalNauseaPercentage: number,
    polysyllables: number,
    readability: Readability,
    semanticCore: SemanticCore[],
    words: SemanticCore[],
    stopWords: SemanticCore[]
}

File: TextAnalyser/Models/SemanticModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TextAnalyser.Models
{
    public class SemanticModel
    {
        public int CharCount { get; set; }
        public int CharCountWithoutSpaces { get; set; }
        public int WordCount { get; set; }
        public int UnicWordCount { get; set; }
        public int SenctenceCount { get; set; }
        public int SignificantWordCount { get; set; }
        public int StopWordCount { get; set; }
        public double WaterPercentage { get; set; }
        public int GrammaticalErrorsCount { get; set; }
        public double ClasicNauseaPercentage { get; set; }
        public double AcademicalNauseaPercentage { get; set; }
        public int Polysyllables { get; set; }
        public Readability Readability { get; set; }
        public List<SemanticCore> SemanticCore { get; set; }
        public List<SemanticCore> Words { get; set; }
        public List<SemanticCore> StopWords { get; set; }
    }
}
